src/arrayMethodReduce.js

```diff
@@ -5,7 +5,19 @@
  */
 function applyCustomReduce() {
   [].__proto__.reduce2 = function(callback, initialValue) {
-    // write code here
+    let result = initialValue;
+    let start = 0;
+
+    if (initialValue === undefined) {
+      result = this[0];
+      start++;
```

Comment:
```suggestion
      start = 1;
```

There's no need to use incrementation here. You need only to values - either 1 or 0

Reply:
Uunderstood